src/main/java/org/colonelkai/guielements/nodes/settings/SettingBox.java

```diff
@@ -0,0 +1,82 @@
+package org.colonelkai.guielements.nodes.settings;
+
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.control.Labeled;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.layout.HBox;
+import javafx.scene.text.Font;
+import org.colonelkai.ForwardLauncher;
+import org.colonelkai.application.Settings;
+
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+
+public class SettingBox<T> extends HBox {
+    private Type type;
+    private T setting;
+    private String displayName;
+    // I feel like this is bad code, and it probably is.
+    PropertyDescriptor propertyDescriptor;
+    Settings temporarySettings;
+
+    Font font = Font.loadFont(
+            ForwardLauncher.class.getResourceAsStream("/fonts/Funkin.otf"), 20);
+
+    public SettingBox(T setting, String displayName, Type type, PropertyDescriptor propertyDescriptor, Settings temporarySettings) throws InvocationTargetException, IllegalAccessException {
+        this.setting = setting;
+        this.displayName = displayName;
+        this.type = type;
+        this.propertyDescriptor = propertyDescriptor;
+        this.temporarySettings = temporarySettings;
+        this.update();
+    }
+
+    public void update() throws InvocationTargetException, IllegalAccessException {
+
+        // ---- LABEL ----
+        Label label = new Label(displayName);
+        label.setFont(font);
+        this.getChildren().add(label);
+
+        // ---- BOOLEAN ----
+        if(this.type == boolean.class) {
```

Comment:
Put this code into its own class. Should have a register which you loop though, when the type matches, run the code

Reply:
So like a class for every type?
A SettingInteractable main class and then ex. BooleanSettingInteractable class that inherits from SettingInteractable? And then I'd have this loop only create that class?

I'm not sure if I understand completely.

Reply:
No no. Just a class that handles the input field, no extension needed